BLUI-3020 As Erice, I want to know how to use icons properly. (#566)
brightlayer-ui/doc-it: src/docs/patterns/icons.mdx

/>

# Icons, Pictograms, and Logos

-   [Image of examples of icons, pictograms, and logos.]

This guideline talks about the design of icons, which symbolizes common actions, objects, or ideas used to aid users. The term “icon” should not be confused with other glyphs such as pictograms and logos.

> Note: if you try to use logos from other companies, make sure that you are compliant with their branding guidelines.

# Icon Usage

## Text labels and Icons

If it is possible to add text labels or use text as an alternative, do not use icons alone. In many situations, texts are better suited to deliver a message accurately, while icons take a lot of guess works. To introduce new icons is like forcing your users to learn a new language. If your application is a mission-critical system, users might even be too afraid to press a button if they do not understand its meaning.

<MaterialDesignDescription
    title={'Icon Usability'}
    url={`https://www.nngroup.com/articles/icon-usability/`}
    description={`Best usability practices for icons`}
    avatar={<LinkIcon style={{ width: 48, height: 48 }} />}
/>

-   [image 1: DON'T: do not use icon buttons when there's plenty of room]

> -   [Image 2: DON'T: do not use icon buttons when their meanings are ambiguous (show monthly test, daily test in a list, where monthly = calendar, daily = clock)]

Author: You even word-smith'd my notes to myself 😮

Icon buttons are primarily used where space is limited, such app bars and lists / tables.

-   [Image 1: App bar with icon buttons]

-   [Image 2: list item with icon buttons, remember to show the "more vert" icon]

-   [Image 3: Text field, showing visibility toggle]

-   [Image 4: a “circle question mark” tooltip, with tooltip explanation of something]

## Inactive Icons inside a component

Inactive Icons are frequently used in components, either as a decorative visual aid to help guide users' attention or to clarify a situation.

-   [Image 1: Successfully created an organization. Show the business logo success state]

-   [Image 2: Icon in a list item. An account preference page where users familiar with the design language but do not understand English can relatively quickly figure out which item corresponds to language change.]

-   [Image 3: A slider, with min and max brightness.]

-   [Image 4: Hero components with icons showing a mix of device parameters]

## Common Universal Icons

A few icons are so widely used these days that they are considered “universal” icons, meaning that they carry the same meaning no matter which application they are used in.

> -   Hamburger, three-dot, chevron, full screens, waterfall, settings, close, back, search, edit

Author: These will be images, yes.

-   Each uses their own images

There are also a few icons commonly seen among Brightlayer UI applications, but less frequently seen elsewhere.

-   Show a few other less obvious ones but still commonly used at Eaton
    -   Logs, warning, error, star (favorite locations), device (assets), all apps 9 dots thing for software suite, toggle visibility of something (pwd), locations
    -   A list, click to expand?
    -   Maybe an option to "edit list on GitHub" for this list

One should never use a universal icon in a distorted way that mismatches its original meaning.

-   [Image 1: a confusing use of hamburger and waterfall icon.]

-   [Image 2: a confusing use of settings and “service” / “repair”]

## Common device icons

Brightlayer UI offers a wide variety of icons for industrial equipment-related icons. Some of them are designed based on their function, while others are designed based on how the device itself looks. We also offer symbols commonly used in one-line diagrams (link to the two NPM packages; Do we want to call those out?).

> While we try our best to add new icons as soon as we identify a need, we noticed that many concepts are very closely related to each other, and it is often unwise to use icons for every single parameter in every single device, while expecting users to distinguish them immediately.

Reviewer: May want to move this down into the section you have below about single purpose icons and do a "See below about single-purpose icons" here.

> While we try our best to add new icons as soon as we identify a need, we noticed that many concepts are very closely related to each other, and it is often unwise to use icons for every single parameter in every single device, while expecting users to distinguish them immediately.

Author: I am keeping it in both sections with a deep link you suggested. I've seen this done wrong so many times that it is worth mentioning twice.

-   [Image 1: AVOID: a list of devices, with a new icon for every single device]

-   [Image 2: use the same icon for the same type of device]
    Icon Variations
    Multi-Colored Icons

WCAG 2.2 AA level requires that non-textual contents of the webpage must achieve a minimum color contrast of 3:1. However, yellow icons are used for warnings and often have a contrast ratio of less than 2:1, therefore failing this requirement. A secondary color would make this icon more legible.

-   [Image: show a warning icon, with a background in yellow and foreground in brown]

## Filled and Outlined Icons

> Brightlayer UI applications may choose from two styles of icons: filled and outlined. You must ensure your icon style stays consistent through the application

Author: No, there's a mix of outlined and filled.

> Brightlayer UI applications may choose from two styles of icons: filled and outlined. You must ensure your icon style stays consistent through the application

Author: I re-worded this part... I hope it is closer to reality.

-   [Show an example of a dashboard list. All icons in a list should be either all filled or all outlined.]

You also have the option to use filled and outlined icons to indicate state. In this case, a filled icon indicates that the item is active, while an outlined icon indicates that the item is inactive.

-   [Show an example of a hero banner, with different statuses. If the count is 0, make it inactive; if it is a positive number, make it active]

## Progress Icons

Progress icons are used as a mini data visualization to display percentages over a whole. It can be the remaining device's health, battery life percentage, etc.

-   [Show examples of percentage icons used in a real application]

Brightlayer currently offers the following progress icons to its designers and developers:

> <ProgressIconCard />

Reviewer: Do we want to remove this from the Iconography page?

Author: This should get removed in a separate PR. Right now this icons guide page is hidden, because I don't know where to put it yet. Once this icon guide page goes live, we can take the ProgressIconCard out from the other side.

## Existing Icon Repositories

Before you are committed to your artwork, make sure to check existing icon resources. In addition to Material Design's official icon set and BLUI icon set, materialdesignicons.com is a community-driven icon repository that follows Material Design's icon guidelines.

You are welcome to take design inspirations from other sources like Font Awesome or follow regulation documents such as those from ANSI and ISO. You are also encouraged to refer to your competitors and see if they have any established conventions. However, you should follow the style guide, as explained below.

## Follow the Style Guide

<MaterialDesignDescription
    title={'System icons'}
    url={'https://material.io/design/iconography/system-icons.html#'}
    description={`Material Design's guidelines on creating your own icons.`}
/>

> You must follow the existing Material Design style guide when creating your own icons. You may get inspired from other sources, but you are required to alter the style so that they fit with existing BLUI icons.

Reviewer: ```suggestion
You must follow the existing Material Design style guide when creating your own icons. You may take inspiration from other sources, but you must update the style so that they fit with existing BLUI icons.
```
